### ISSUE TITLE
Latest cucumber-core jar has dependency to io.cucumber:messages:18.0.0 with CVE issue

### ISSUE THREAD

**Issue by topsycreed** [cucumber/cucumber-jvm#2599]
### 👓 What did you see?
A CVE issue in io.cucumber:messages:18.0.0 dependency if we are adding io.cucumber:cucumber-junit or io.cucumber:cucumber-java with latest version ("7.5.0"). 
If I will add a new version of io.cucumber:messages:19.1.2 explicitly then I cannot run my tests and see error:
`TestRunner > initializationError FAILED
build	04-Aug-2022 17:30:26	    java.lang.NoSuchMethodError: 'void io.cucumber.messages.types.Step.<init>(io.cucumber.messages.types.Location, java.lang.String, java.lang.String, io.cucumber.messages.types.DocString, io.cucumber.messages.types.DataTable, java.lang.String)'
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.GherkinDocumentBuilder.getTransformedNode(GherkinDocumentBuilder.java:87)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.GherkinDocumentBuilder.endRule(GherkinDocumentBuilder.java:73)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser$3.call(Parser.java:198)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser$3.call(Parser.java:196)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.handleExternalError(Parser.java:166)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.handleAstError(Parser.java:161)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.endRule(Parser.java:196)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.matchTokenAt_15(Parser.java:1610)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.matchToken(Parser.java:369)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.parse(Parser.java:136)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.parse(Parser.java:109)
build	04-Aug-2022 17:30:26	        at io.cucumber.gherkin.Parser.parse(Parser.java:105)
error	04-Aug-2022 17:30:26	`

### ✅ What did you expect to see?
That core library has a dependency to a latest messages and it's working correctly.

**Comment by mpkorstanje** [comment]
Could you please explain which CVE and how it might impact you?

I presume you are looking at CVE-2020-36518, however Jackson is only used as a test scoped dependency as you can see by inspecting the [messages/pom.xml](https://github.com/cucumber/common/blob/messages/v18.0.0/messages/java/pom.xml). You should not be pulling in Jackson as a dependency though Cucumber.

**Comment by chushkin** [comment]
So, even upgrading to the latest cucumber version is still vulnerable, as cucumber has not upgraded messages jar reference in their framework beyond 18.0.0
CVE-2021-22569 - for google protobuf
CVE-2022-25647 - for gson
According to our Prisma scan, these are in the messages jar version 18.0.0
According to maven, that jar is also vulnerable

**Comment by mpkorstanje** [comment]
> So, even upgrading to the latest cucumber version is still vulnerable, as cucumber has not upgraded messages jar reference in their framework beyond 18.0.0

You may not be on the latest version.

```
$ git clone https://github.com/cucumber/cucumber-java-skeleton
$ cd cucumber-java-skeleton
$ mvn dependency:tree
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< cucumber:cucumber-java-skeleton >-------------------
[INFO] Building Cucumber-Java Skeleton 0.0.1
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ cucumber-java-skeleton ---
[INFO] cucumber:cucumber-java-skeleton:jar:0.0.1
[INFO] +- io.cucumber:cucumber-java:jar:7.6.0:test
[INFO] |  +- io.cucumber:cucumber-core:jar:7.6.0:test
[INFO] |  |  +- io.cucumber:cucumber-gherkin:jar:7.6.0:test
[INFO] |  |  +- io.cucumber:cucumber-gherkin-messages:jar:7.6.0:test
[INFO] |  |  |  \- io.cucumber:gherkin:jar:24.0.0:test
[INFO] |  |  +- io.cucumber:messages:jar:19.0.0:test
[INFO] |  |  +- io.cucumber:tag-expressions:jar:4.1.0:test
[INFO] |  |  +- io.cucumber:cucumber-expressions:jar:16.0.0:test
[INFO] |  |  +- io.cucumber:datatable:jar:7.6.0:test
[INFO] |  |  +- io.cucumber:cucumber-plugin:jar:7.6.0:test
[INFO] |  |  +- io.cucumber:docstring:jar:7.6.0:test
[INFO] |  |  +- io.cucumber:html-formatter:jar:20.0.0:test
[INFO] |  |  \- io.cucumber:ci-environment:jar:9.1.0:test
[INFO] |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test
[INFO] +- io.cucumber:cucumber-junit-platform-engine:jar:7.6.0:test
[INFO] |  \- org.junit.platform:junit-platform-engine:jar:1.9.0:test
[INFO] |     +- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |     \- org.junit.platform:junit-platform-commons:jar:1.9.0:test
[INFO] +- org.junit.platform:junit-platform-suite:jar:1.9.0:test
[INFO] |  +- org.junit.platform:junit-platform-suite-api:jar:1.9.0:test
[INFO] |  \- org.junit.platform:junit-platform-suite-engine:jar:1.9.0:test
[INFO] |     \- org.junit.platform:junit-platform-suite-commons:jar:1.9.0:test
[INFO] |        \- org.junit.platform:junit-platform-launcher:jar:1.9.0:test
[INFO] \- org.junit.jupiter:junit-jupiter:jar:5.9.0:test
[INFO]    +- org.junit.jupiter:junit-jupiter-api:jar:5.9.0:test
[INFO]    +- org.junit.jupiter:junit-jupiter-params:jar:5.9.0:test
[INFO]    \- org.junit.jupiter:junit-jupiter-engine:jar:5.9.0:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.733 s
[INFO] Finished at: 2022-08-08T19:57:41+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by chushkin** [comment]
Here is the 7.5.0 link for that, to explain what we were seeing and how: https://mvnrepository.com/artifact/io.cucumber/cucumber-core/7.5.0
and core uses messages: https://mvnrepository.com/artifact/io.cucumber/messages/18.0.0
which is vulnerable

But I see that yesterday 7.6.0 became GA and is using messages 19.0.0 (which is not vulnerable)
So, we just need to upgrade to 7.6.0

new version https://mvnrepository.com/artifact/io.cucumber/cucumber-core/7.6.0 uses https://mvnrepository.com/artifact/io.cucumber/messages/19.0.0
I think no more issue for us

**Comment by mpkorstanje** [comment]
> Here is the 7.5.0 link for that, to explain what we were seeing and how: https://mvnrepository.com/artifact/io.cucumber/cucumber-core/7.5.0
and core uses messages: https://mvnrepository.com/artifact/io.cucumber/messages/18.0.0
which is vulnerable

Cheers. Unfortunately that is a false positive. As you can see on the page, Jackson is a test scoped dependency and not part of the distributed artifact. Without further explanation of why your tooling considers vulnerabilities in test scoped dependencies a problem (for you) it would appear to be a little paranoid.

Regardless, looks like this is no longer relevant as it is resolved in v7.6.

**Comment by chushkin** [comment]
Thank you
